packages/x-date-pickers/src/internals/components/FakeTextField/FakeTextField.tsx

```diff
@@ -1,68 +1,131 @@
-  content: React.HTMLAttributes<HTMLSpanElement>;
-  before: React.HTMLAttributes<HTMLSpanElement>;
-  after: React.HTMLAttributes<HTMLSpanElement>;
-}
+const FakeTextFieldRoot = styled(FormControl, {
+  name: 'MuiFakeTextField',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})({});
 
-interface FakeTextFieldProps {
-  elements: FakeTextFieldElement[];
-  valueStr: string;
-  onValueStrChange: React.ChangeEventHandler<HTMLInputElement>;
-  error: boolean;
-  id?: string;
-  InputProps: any;
-  inputProps: any;
-  disabled?: boolean;
-  autoFocus?: boolean;
-  ownerState?: any;
-  valueType: 'value' | 'placeholder';
-}
+const useUtilityClasses = (ownerState: FakeTextFieldProps) => {
+  const { focused, disabled, classes, required } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      focused && !disabled && 'focused',
+      disabled && 'disabled',
+      required && 'required',
+    ],
+  };
+
+  return composeClasses(slots, getFakeTextFieldUtilityClass, classes);
+};
 
 export const FakeTextField = React.forwardRef(function FakeTextField(
   props: FakeTextFieldProps,
   ref: React.Ref<HTMLDivElement>,
 ) {
   const {
     elements,
+    className,
+    color = 'primary',
+    disabled = false,
+    error = false,
+    label,
+    variant = 'outlined',
+    fullWidth = false,
     valueStr,
-    onValueStrChange,
-    id,
-    error,
-    InputProps,
-    inputProps,
-    autoFocus,
-    disabled,
+    helperText,
     valueType,
-    ownerState,
+    id: idOverride,
+    FormHelperTextProps,
+    InputLabelProps,
+    inputProps,
+    InputProps,
+    required = false,
+    focused: focusedProp,
+    ownerState: ownerStateProp,
     ...other
   } = props;
 
+  const [focused, setFocused] = React.useState(focusedProp);
+
+  const rootRef = React.useRef<HTMLDivElement>(null);
+  const handleRootRef = useForkRef(ref, rootRef);
+
+  const inputRef = React.useRef<HTMLDivElement>(null);
+  const handleInputRef = useForkRef(inputRef, InputProps?.ref);
+
+  const id = useId(idOverride);
+  const helperTextId = helperText && id ? `${id}-helper-text` : undefined;
+  const inputLabelId = label && id ? `${id}-label` : undefined;
+
+  const ownerState = {
+    ...props,
+    color,
+    disabled,
+    error,
+    focused,
+    variant,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  // TODO: delete after behavior implementation
+  const onWrapperClick = () => {
+    if (!focused) {
+      setFocused(true);
+      const container = rootRef.current;
+
+      // Find the first input element within the container
+      const firstInput = container?.querySelector<HTMLElement>('.content');
+
+      // Check if the input element exists before focusing it
+      if (firstInput) {
+        firstInput.focus();
+      }
+    }
+  };
+
   return (
-    <React.Fragment>
-      <Box
-        ref={ref}
+    <FakeTextFieldRoot
+      className={clsx(classes.root, className)}
+      ref={handleRootRef}
+      {...{
```

Comment:
Is there a benefit to spread here instead of just passing props as usual?


```suggestion
      focused={focused}
      // ....etc...
```

Reply:
Just a personal preference I guess. Feels cleaner to me, but I can change it. 👍